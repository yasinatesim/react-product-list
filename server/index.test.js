const rp = require('request-promise')

const API = 'http://localhost:4000/';

describe('Getting shop items', () => {

  it('Retrieves the shop items in default order', async () => {
    const query = `
    query {
    products {
      id
      name
      image
      price
      url
      liked
      cargo {
        type
        name
      }
    }
  }`;

    const response = await rp({ method: 'POST', uri: API, body: { query }, json: true });
    expect(response).toMatchSnapshot();
  });
});
