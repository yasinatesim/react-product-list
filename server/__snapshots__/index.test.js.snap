// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getting shop items Retrieves the shop items in default order 1`] = `
Object {
  "data": Object {
    "products": Array [
      Object {
        "cargo": Object {
          "name": "Ücretsiz Kargo",
          "type": 1,
        },
        "id": "43447503",
        "image": "https://cdn.dsmcdn.com/ty6/product/media/images/20200716/11/4596645/77043792/0/0_org_zoom.jpg",
        "liked": false,
        "name": "Akıllı Saat B08 Smart Watch Arama Cevaplama Sos Tam Dokunmatik Pembe",
        "price": 228,
        "url": "https://www.trendyol.com/koctech/akilli-saat-b08-smart-watch-arama-cevaplama-sos-tam-dokunmatik-pembe-p-43447503?boutiqueId=518609&merchantId=184171",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz Kargo",
          "type": 1,
        },
        "id": "35132628",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20200128/13/2588250/62725467/1/1_org_zoom.jpg",
        "liked": false,
        "name": "LS 01 (Siyah)Global Versiyon Akıllı Saat",
        "price": 259,
        "url": "https://www.trendyol.com/haylou/ls-01-siyahglobal-versiyon-akilli-saat-p-35132628?boutiqueId=523669&merchantId=104845",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "36301871",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20200226/14/3689875/64604656/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Wmax GW600S Suya Dayanıklı Akıllı Çocuk Takip Saati",
        "price": 270,
        "url": "https://www.trendyol.com/vmax/wmax-gw600s-suya-dayanikli-akilli-cocuk-takip-saati-p-36301871?boutiqueId=523184&merchantId=144894",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "37813529",
        "image": "https://cdn.dsmcdn.com/ty2/product/media/images/20200331/22/825986/67214424/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Smart Round Akıllı Saat Siyah   Pssw06",
        "price": 99,
        "url": "https://www.trendyol.com/polosmart/smart-round-akilli-saat-siyah-pssw06-p-37813529?boutiqueId=523669&merchantId=110480",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "32070306",
        "image": "https://cdn.dsmcdn.com/ty6/product/media/images/20200717/19/5160047/57641439/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Galaxy Watch Active2 R830 Pink Gold 40mm",
        "price": 1598,
        "url": "https://www.trendyol.com/samsung/galaxy-watch-active2-r830-pink-gold-40mm-p-32070306?boutiqueId=523669&merchantId=106912",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "32217109",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20191024/16/470245/57901760/1/1_org_zoom.jpg",
        "liked": false,
        "name": "M26 Akıllı Saat Smart Watch Ios-android Uyumlu Siyah",
        "price": 77,
        "url": "https://www.trendyol.com/polygold/m26-akilli-saat-smart-watch-ios-android-uyumlu-siyah-p-32217109?boutiqueId=523669&merchantId=116895",
      },
      Object {
        "cargo": Object {
          "name": "Ücretli Kargo",
          "type": 3,
        },
        "id": "36984752",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20200318/10/4516355/65809243/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Smart Bracelet St1  Akıllı Saat Su Geçirmez",
        "price": 282,
        "url": "https://www.trendyol.com/kingboss/smart-bracelet-st1-akilli-saat-su-gecirmez-p-36984752?boutiqueId=523847&merchantId=107885",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "32897645",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20191119/17/680985/59221658/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Dz-09 Sim Kart Destekli Türkçe Menü Kamerali Akıllı Saat",
        "price": 120,
        "url": "https://www.trendyol.com/3mk/dz-09-sim-kart-destekli-turkce-menu-kamerali-akilli-saat-p-32897645?boutiqueId=522747&merchantId=186308",
      },
      Object {
        "cargo": Object {
          "name": "Ücretli Kargo",
          "type": 3,
        },
        "id": "7026402",
        "image": "https://cdn.dsmcdn.com/Assets/ProductImages/oa/53/7026402/1/190198806185_1_org_zoom.jpg",
        "liked": false,
        "name": " Watch MTF02TU/A Seri 3 GPS, 38mm Space Grey Aluminium Case with Black Sport Band",
        "price": 1749,
        "url": "https://www.trendyol.com/apple/watch-mtf02tu-a-seri-3-gps-38mm-space-grey-aluminium-case-with-black-sport-band-p-7026402?boutiqueId=523669&merchantId=113912",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "43839081",
        "image": "https://cdn.dsmcdn.com/ty7/product/media/images/20200721/4/5407387/77694636/0/0_org_zoom.jpg",
        "liked": false,
        "name": "Akıllı Çocuk Saati Konum Takipli Sim Kart Girişli Lbs Ve Sos Özellikli-lila",
        "price": 250,
        "url": "https://www.trendyol.com/smartberry/akilli-cocuk-saati-konum-takipli-sim-kart-girisli-lbs-ve-sos-ozellikli-lila-p-43839081?boutiqueId=523098&merchantId=181635",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz Kargo",
          "type": 1,
        },
        "id": "37363051",
        "image": "https://cdn.dsmcdn.com/assets/dev/product/media/images/20200325/10/732/66375196/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Olx Suya Dayanıklı Takip Özellikli Kameralı Imei Kayıtlı Sim Kartlı Akıllı Çocuk Saati Pembe",
        "price": 245,
        "url": "https://www.trendyol.com/king/olx-suya-dayanikli-takip-ozellikli-kamerali-imei-kayitli-sim-kartli-akilli-cocuk-saati-pembe-p-37363051?boutiqueId=523847&merchantId=144894",
      },
      Object {
        "cargo": Object {
          "name": "Ücretli Kargo",
          "type": 3,
        },
        "id": "41771295",
        "image": "https://cdn.dsmcdn.com/ty8/product/media/images/20200724/1/5784450/74129046/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Akıllı Pembe Çocuk Takip Saati Sim Kartlı Gpsli Btk Kayıtlı Kameralı",
        "price": 233.99,
        "url": "https://www.trendyol.com/fabby/akilli-pembe-cocuk-takip-saati-sim-kartli-gpsli-btk-kayitli-kamerali-p-41771295?boutiqueId=504991&merchantId=122474",
      },
      Object {
        "cargo": Object {
          "name": "Ücretli Kargo",
          "type": 3,
        },
        "id": "32185904",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20191023/16/466655/57847516/1/1_org_zoom.jpg",
        "liked": false,
        "name": "U8 Akıllı Saat Smart Watch Ios Ve Android Uyumlu Bluetooth Saat Kırmızı",
        "price": 61,
        "url": "https://www.trendyol.com/polygold/u8-akilli-saat-smart-watch-ios-ve-android-uyumlu-bluetooth-saat-kirmizi-p-32185904?boutiqueId=523669&merchantId=116895",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz Kargo",
          "type": 1,
        },
        "id": "43310261",
        "image": "https://cdn.dsmcdn.com/ty5/product/media/images/20200714/15/4466299/76835116/0/0_org_zoom.jpg",
        "liked": false,
        "name": "Q530/2020 Sim Kartlı Akıllı Çocuk Saati - Pembe",
        "price": 270,
        "url": "https://www.trendyol.com/smartbell/q530-2020-sim-kartli-akilli-cocuk-saati-pembe-p-43310261?boutiqueId=518609&merchantId=114471",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz Kargo",
          "type": 1,
        },
        "id": "42195805",
        "image": "https://cdn.dsmcdn.com/ty6/product/media/images/20200620/22/3257111/74912039/0/0_org_zoom.jpg",
        "liked": false,
        "name": "Akıllı Saat T500 Smart Watch Türkçe Menü Tam Dokunmatik - Pudra",
        "price": 249.99,
        "url": "https://www.trendyol.com/fabby/akilli-saat-t500-smart-watch-turkce-menu-tam-dokunmatik-pudra-p-42195805?boutiqueId=507942&merchantId=122474",
      },
      Object {
        "cargo": Object {
          "name": "Ücretli Kargo",
          "type": 3,
        },
        "id": "36468436",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20200304/15/4043331/64911704/1/1_org_zoom.jpg",
        "liked": false,
        "name": "W4 Akıllı Saat Smart Watch Bluetooth Dokunmatik Su Geçirmez Ip67",
        "price": 163.98,
        "url": "https://www.trendyol.com/thorqtech/w4-akilli-saat-smart-watch-bluetooth-dokunmatik-su-gecirmez-ip67-p-36468436?boutiqueId=507942&merchantId=159065",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz Kargo",
          "type": 1,
        },
        "id": "37594546",
        "image": "https://cdn.dsmcdn.com/ty10/product/media/images/20200828/21/9030048/66830937/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Akıllı Çocuk Saati Gps Takip Özellikli Kameralı Sim Kartlı Saat 2030 Model Smart Watch",
        "price": 242,
        "url": "https://www.trendyol.com/smart-way/akilli-cocuk-saati-gps-takip-ozellikli-kamerali-sim-kartli-saat-2030-model-smart-watch-p-37594546?boutiqueId=525395&merchantId=181635",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "32072113",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20191018/14/440004/57645929/1/1_org_zoom.jpg",
        "liked": false,
        "name": "M3 Band Spor Modları, Nabız Ölçer Akıllı Bileklik Akıllı Saat",
        "price": 51.3,
        "url": "https://www.trendyol.com/angel-eye/m3-band-spor-modlari-nabiz-olcer-akilli-bileklik-akilli-saat-p-32072113?boutiqueId=523669&merchantId=128663",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "32073294",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20191018/15/440082/57646201/1/1_org_zoom.jpg",
        "liked": false,
        "name": "H1 Smart Watch H1 Bayan Akıllı Saat Nabız Ölçer Uyku Ve Spor Faaliyetleri Gold Renk",
        "price": 279,
        "url": "https://www.trendyol.com/kingshark/h1-smart-watch-h1-bayan-akilli-saat-nabiz-olcer-uyku-ve-spor-faaliyetleri-gold-renk-p-32073294?boutiqueId=448097&merchantId=118796",
      },
      Object {
        "cargo": Object {
          "name": "Ücretli Kargo",
          "type": 3,
        },
        "id": "38065000",
        "image": "https://cdn.dsmcdn.com/ty1/product/media/images/20200403/13/1086264/67609206/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Smart Round Akıllı Saat Mavi  Pssw06",
        "price": 99,
        "url": "https://www.trendyol.com/polosmart/smart-round-akilli-saat-mavi-pssw06-p-38065000?boutiqueId=523669&merchantId=110480",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "35904945",
        "image": "https://cdn.dsmcdn.com/assets/product/media/images/20200224/17/3455934/64153751/1/1_org_zoom.jpg",
        "liked": false,
        "name": "Amazfit A1914 GTS Amoled Ekran 5ATM Suya Dayanıklı Akıllı Saat - Lava Gri",
        "price": 798,
        "url": "https://www.trendyol.com/amazfit/amazfit-a1914-gts-amoled-ekran-5atm-suya-dayanikli-akilli-saat-lava-gri-p-35904945?boutiqueId=523669&merchantId=110994",
      },
      Object {
        "cargo": Object {
          "name": "Ücretsiz – Aynı Gün Kargo",
          "type": 2,
        },
        "id": "42539590",
        "image": "https://cdn.dsmcdn.com/ty6/product/media/images/20200630/0/3641140/75561707/0/0_org_zoom.jpg",
        "liked": false,
        "name": "Nln-q19 Sim Kartlı Akıllı Çocuk Saati Gps Takip - Yeşil",
        "price": 249,
        "url": "https://www.trendyol.com/divex/nln-q19-sim-kartli-akilli-cocuk-saati-gps-takip-yesil-p-42539590?boutiqueId=512155&merchantId=181635",
      },
    ],
  },
}
`;
